lib/endpoint.ex

```diff
@@ -73,9 +73,9 @@ defmodule SparkPost.Endpoint do
   defp handle_response({:ok, %HTTPoison.Response{status_code: code, body: body}}, decode_results) when code >= 200 and code < 300 do
     decoded_body = decode_response_body(body)
     if decode_results do
-      %SparkPost.Endpoint.Response{status_code: 200, results: decoded_body.results}
+      %SparkPost.Endpoint.Response{status_code: code, results: decoded_body.results}
```

Comment:
Will have to check which other modules either make assumptions about the status code or pass it on to client code. This could produce a (good) breaking change in those cases.